src/Tribe/Tickets.php

```diff
@@ -2110,6 +2110,324 @@ public function generate_tickets_email_content( $tickets ) {
+			$provider             = $args['provider'];
+			$post_id              = $args['post_id'];
+			$order_id             = $args['order_id'];
+			$ticket_sent_meta_key = $args['ticket_sent_meta_key'];
+			$send_callback        = $args['send_callback'];
+
+			// If invalid send callback, do not send the email.
+			if ( ! is_callable( $send_callback ) ) {
+				return false;
+			}
+
+			// Set up default content.
+			if ( empty( $content ) ) {
+				$content = $this->generate_tickets_email_content( $tickets );
+			}
+
+			// Set up default subject.
+			if ( empty( $subject ) ) {
+				$site_name = stripslashes_deep( html_entity_decode( get_bloginfo( 'name' ), ENT_QUOTES ) );
+				$is_rsvp   = 'rsvp' === $provider;
+
+				$singular = $is_rsvp
+					? tribe_get_rsvp_label_singular( 'RSVP email send' )
+					: tribe_get_ticket_label_singular_lowercase( 'ticket email send' );
+
+				$plural = $is_rsvp
+					? tribe_get_rsvp_label_plural( 'RSVPs email send' )
+					: tribe_get_ticket_label_plural_lowercase( 'tickets email send' );
+
+				// translators: %1$s: The singular of "RSVP" or "ticket", %2$s: The plural of "RSVPs" or "tickets", %3$s: The site name.
+				$subject_string = _nx( 'Your %1$s from %3$s', 'Your %2$s from %3$s', count( $tickets ), 'The default RSVP/ticket email subject', 'event-tickets' );
+
+				$subject = sprintf(
+					$subject_string,
+					$singular,
+					$plural,
+					$site_name
+				);
+			}
+
+			// Add From name/email to headers if no headers set yet and we have a valid From email address.
+			if ( empty( $headers ) && ! empty( $from_name ) && ! empty( $from_email ) && is_email( $from_email ) ) {
+				$from_email = filter_var( $from_email, FILTER_SANITIZE_EMAIL );
+
+				$headers[] = sprintf(
+					'From: %1$s <%2$s>',
+					filter_var( $from_name, FILTER_SANITIZE_STRING ),
+					$from_email
+				);
+
+				$headers[] = sprintf(
+					'Reply-To: %s',
+					$from_email
+				);
+			}
+
+			// Enforce headers array.
+			if ( ! is_array( $headers ) ) {
+				$headers = explode( "\r\n", $headers );
+			}
+
+			// Enforce text/html content type header.
+			if ( ! in_array( 'Content-type: text/html', $headers, true ) || ! in_array( 'Content-type: text/html; charset=utf-8', $headers, true ) ) {
+				$headers[] = 'Content-type: text/html; charset=utf-8';
+			}
+
+			/**
+			 * Allow filtering the email recipient for a provider. Backwards compatible with previous provider filter.
+			 *
+			 * The dynamic portion of the filter hook, `$provider`, refers to the provider slug (rsvp, tpp, woo, edd).
+			 *
+			 * @since TBD
+			 *
+			 * @param string     $to       The email to send to.
+			 * @param int        $post_id  The post/event ID to send the email for.
+			 * @param string|int $order_id The order ID to send the email for.
+			 * @param array      $tickets  The list of tickets to send.
+			 */
+			$to = apply_filters( "tribe_{$provider}_email_recipient", $to, $post_id, $order_id, $tickets );
```

Comment:
Maybe add the `_tickets` prefix before provider to keep in line with things?

Reply:
This is just backcompat for the old filter names :( not much I could do there unfortunately, but we can remove them once we start recommending the better ones.